### PR TITLE
Make compatible with older versions of python

### PR DESCRIPTION
**Pull request by nbird11**
Avoid the use of f-strings and variable typing and new syntax for parameter defaults.

### CODE REVIEW

File: bird-slideshow.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 from bs4 import BeautifulSoup
 
 class Config:
-    def __init__(self, config_file: str | None) -> None:
```
Comment (tbird20d) [inline bird-slideshow.py:20]: Do you know what version of python supports this syntax?

File: bird-slideshow.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
...
+    def __init__(self, config_file=None):
```
Comment (tbird20d) [inline bird-slideshow.py:20]: This looks good.  Thanks.

File: bird-slideshow.py [hunk 3]
```diff
@@ -240,7 +240,7 @@ def download_img(cache_dir, img_link):
         return filepath
 
     except:
-        print("Error", f"Could not download {img_link}")
+        print("Error", "Could not download %s" % img_link)
```
Comment (tbird20d) [inline bird-slideshow.py:243]: You could use a + here to avoid having to interpolate the string:

ex: "Could not download " + img_link

Comment (nbird11) [reply bird-slideshow.py:243]: %s interpolation is more consistent with the rest of the program.

File: bird-slideshow.py (hunk 2 above)
Comment (nbird11) [reply bird-slideshow.py:20]: Vertical Bar was introduced for Union types in 3.10

File: bird-slideshow.py (hunk 1 above)
Comment (nbird11) [reply bird-slideshow.py:20]: Vertical Bar was introduced for Union types in 3.10

File: bird-slideshow.py (hunk 3 above)
Comment (nbird11) [reply bird-slideshow.py:243]: %s interpolation is more consistent with the rest of the program.

**Pull request merged**